cloud-operations/packer-image-builder/main.tf

```diff
@@ -0,0 +1,129 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  compute_subnet_name = "image-builder"
+  compute_zone        = "${var.region}-a"
+}
+
+module "project" {
+  source          = "../../modules/project"
+  name            = var.project_id
+  parent          = try(var.root_node, null)
+  billing_account = try(var.billing_account, null)
+  project_create  = var.project_create
+  services = [
+    "compute.googleapis.com"
+  ]
+  service_config = {
+    disable_on_destroy         = false
+    disable_dependent_services = false
+  }
+}
+
+module "service-account-image-builder" {
+  source     = "../../modules/iam-service-account"
+  project_id = module.project.project_id
+  name       = "image-builder"
+  iam_project_roles = {
+    (var.project_id) = [
+      "roles/compute.instanceAdmin.v1",
+      "roles/iam.serviceAccountUser"
+    ]
+  }
+}
+
+module "service-account-image-builder-vm" {
+  source     = "../../modules/iam-service-account"
+  project_id = module.project.project_id
+  name       = "image-builder-vm"
+}
+
+module "vpc" {
+  source     = "../../modules/net-vpc"
+  project_id = module.project.project_id
+  name       = "image-builder"
+  subnets = [
+    {
+      name               = local.compute_subnet_name
+      ip_cidr_range      = var.cidrs.image-builder
+      region             = var.region
+      secondary_ip_range = null
+    }
+  ]
+}
+
+module "firewall" {
+  source     = "../../modules/net-vpc-firewall"
+  project_id = module.project.project_id
+  network    = module.vpc.name
+  custom_rules = {
+    image-builder-ingress-builder-vm = {
+      description          = "Allow image builder vm ingress traffic"
+      direction            = "INGRESS"
+      action               = "allow"
+      sources              = []
+      ranges               = var.packer_source_cidrs
+      targets              = [module.service-account-image-builder-vm.email]
+      use_service_accounts = true
+      rules                = [{ protocol = "tcp", ports = [22, 5985, 5986] }]
+      extra_attributes     = {}
+    }
+  }
+}
+
+module "nat" {
+  source                = "../../modules/net-cloudnat"
+  project_id            = module.project.project_id
+  region                = var.region
+  name                  = "default"
+  router_network        = module.vpc.name
+  config_source_subnets = "LIST_OF_SUBNETWORKS"
+  subnetworks = [
+    {
+      self_link            = module.vpc.subnet_self_links["${var.region}/${local.compute_subnet_name}"]
+      config_source_ranges = ["ALL_IP_RANGES"]
+      secondary_ranges     = null
+    }
+  ]
+}
+
+resource "google_service_account_iam_binding" "sa-image-builder-token-creators" {
+  count              = length(var.packer_account_users) > 0 ? 1 : 0
```

Comment:
Can we use `for_each` here?

Reply:
Doable, but I will have to use `google_service_account_iam_member` instead of binding. 
In case of binding, we want to have only one with a list of members from `var.packer_account_users`. Use of binding makes sense here for me.

Reply:
You could do `for_each = toset(length(var.packer_account_users) > 0 ? [] : [""])`, but I see your point. :)

Reply:
or `for_each = length(var.packer_account_users) > 0 ? {} : {1=1}` (extra points for longer substring with non-alphabetic characters ;) )